src/deployconfig.jl

```diff
@@ -843,6 +843,200 @@ authentication_method(::Buildkite) = Documenter.SSH
 
 documenter_key(::Buildkite) = ENV["DOCUMENTER_KEY"]
 
+#################
+# Woodpecker CI #
+#################
+
+"""
+    Woodpecker <: DeployConfig
+
+Implementation of `DeployConfig` for deploying from Woodpecker CI.
+
+The following environmental variables are built-in from the Woodpecker pipeline
+influences how `Documenter` works:
+ - `CI_REPO`: must match the full name of the repository <owner>/<name> e.g. `JuliaDocs/Documenter.jl`
+ - `CI_BUILD_EVENT`: must be set to `push`, `tag`, `pull_request`, and `deployment`
+ - `CI_COMMIT_REF`: must match the `devbranch` keyword to [`deploydocs`](@ref), alternatively correspond to a git tag.
```

Comment:
`CI_COMMIT_PULL_REQUEST`, `CI_REPO_LINK` and `CI_REPO_OWNER` should also probably be here? And while you do mention `FORGE_URL` and `PROJECT_ACCESS_TOKEN` below, it might be good to have them in this list too, for completeness.